test/components/legendTests.ts

```diff
@@ -254,6 +254,46 @@ describe("Legend", () => {
     svg.remove();
   });
 
+  it("can set formatter and change how entry labels are displayed", () => {
+    color.domain(["jtao", "mschafer", "kfalter"]);
+    let data: {[id: string]: string; } = {"jtao": "Joy", "mschafer":  "Mark", "kfalter": "Kelsey"};
+    let formatter = (id: string) => data[id];
+    legend.formatter(formatter);
+    legend.renderTo(svg);
+    let legendRows = svg.selectAll(entrySelector);
+    assert.operator(legendRows.size(), ">=", 1, "There is at least one entry in the test");
+    legendRows.each(function(d: Element, i: number){
+      let expectText = formatter(legend.colorScale().domain()[i]);
+      assert.strictEqual(d3.select(this).select("text").text(), expectText,
+        `formatter output ${expectText} should be displayed`);
+    });
+    svg.remove();
+  });
+
+  it("can get formatter of the legend using formatter()", () => {
+    color.domain(["jtao", "mschafer", "kfalter"]);
+    let data: {[id: string]: string; } = {"jtao": "Joy", "mschafer": "Mark", "kfalter": "Kelsey"};
+    let formatter = (id: string) => data[id];
+    legend.formatter(formatter);
+    assert.strictEqual(legend.formatter(), formatter, "formatter() returns the formatter of legend correctly");
+    svg.remove();
+  });
+
+  it("comparator() works as expected when customized formatter is applied", () => {
+    let colorDomain = ["A", "B", "C"];
+    color.domain(colorDomain);
+    let expectedTexts = ["Z", "Y", "X"];
+    let formatter = (d: string) => expectedTexts[colorDomain.indexOf(d)];
+    legend.formatter(formatter);
+    let comparator = (a: string, b: string) => a.localeCompare(b);
+    legend.comparator(comparator);
+    legend.renderTo(svg);
+    let entryTexts = svg.selectAll(entrySelector)[0].map((node: Element) => d3.select(node).select("text").text());
+    expectedTexts.sort(comparator);
```

Comment:
My logic is:
Given a sorted array of domain names and an unsorted array of expected texts, the actual displayed texts should be sorted. So I need to compare the actual displayed texts and sorted expected texts to see whether comparator sorts in the order of displayed texts.